### PR TITLE
Tooltip system

### PR DESCRIPTION
**Pull request by dsamia25**
- Structures now have a TooltipDescription object with a name, description, icon, and list of tooltip actions. 
- Tooltips can take in actions that get added as buttons onto the tooltip when opened. 
- Select a tile with a structure to open the tooltip for that structure.
- Deselect the tile to close the tooltip.
- There is a bug where sometimes selecting a tile with a structure while another tooltip is open will do nothing (the current tooltip will stay open and the new one will not open).

### CODE REVIEW

File: Assets/Scripts/Gameplay/Tile.cs [hunk 1]
```diff
@@ -31,6 +31,9 @@ public enum Anchor
         
         private List<Structure> m_structures;
 
+        // Public viewable list of structures. TODO: The structures in the list are still editable.
+        public List<Structure> Structures => new List<Structure>(m_structures);
```
Comment (ZotChryis) [inline Assets/Scripts/Gameplay/Tile.cs:35]: You can still change the internal structure, so not sure what the point of wrapping this in another list.  Might as well just return m_structures for now. We can make a ReadOnlyList variant for the property, if we're concerned with data security

File: Assets/Scripts/UI/TooltipView.cs [hunk 2]
```diff
@@ -1,21 +1,99 @@
...
 
 namespace UI
 {
     public class TooltipView : View
     {
+        /// <summary>
+        /// Holds all information that would be displayed in a tooltip.
+        /// </summary>
+        [Serializable]
+        public class TooltipDescription
```
Comment (ZotChryis) [inline Assets/Scripts/UI/TooltipView.cs:16]: TooltipData? TooltipInfo? TooltipProperties? Description is overloaded in one of the members, aka `public string description`

File: Assets/Scripts/Schemas/StructureSchema.cs [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 using UnityEngine;
+using UI;
 
 namespace Schemas
 {
     [CreateAssetMenu]
     public class StructureSchema : InvokerSchema
     {
         public Structure Prefab;
+        public TooltipView.TooltipDescription TooltipInfo;  // TODO: Move this somewhere else.
```
Comment (ZotChryis) [inline Assets/Scripts/Schemas/StructureSchema.cs:11]: Perhaps we have something like `ITooltipable` instead?

File: Assets/Scripts/UI/TooltipView.cs (hunk 2 above)
Comment (rsamia) [reply Assets/Scripts/UI/TooltipView.cs:16]: What if we just moved all this data like title, description, sprite, actions into the structure schema, and this TooltipDescription can instead be a struct that is NOT serializable, and we just new one up and pass the data through when creating the tooltip

**Review by rsamia — APPROVED** [review]

File: Assets/Scripts/Gameplay/Tile.cs (hunk 1 above)
Comment (dsamia25) [reply Assets/Scripts/Gameplay/Tile.cs:35]: Made m_structures public to return it instead

**Pull request merged**